src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -293,85 +263,54 @@ internal static TableOperation CreateMetricsInsertOperation(ScaleMetrics metrics
             string rowKey = TableStorageHelpers.GetRowKey(now.Value);
 
             var entity = TableEntityConverter.ToEntity(metrics, hostId, rowKey, metrics.Timestamp);
-            entity.Properties.Add(MonitorIdPropertyName, EntityProperty.GeneratePropertyForString(descriptor.Id));
+            entity.Add(MonitorIdPropertyName, descriptor.Id);
 
             // We map the sample timestamp to its own column so it doesn't conflict with the built in column.
             // We want to ensure that timestamp values for returned metrics are precise and monotonically
             // increasing when ordered results are returned. The built in timestamp doesn't guarantee this.
-            entity.Properties.Add(SampleTimestampPropertyName, EntityProperty.GeneratePropertyForDateTimeOffset(metrics.Timestamp));
+            entity.Add(SampleTimestampPropertyName, metrics.Timestamp);
 
-            return TableOperation.Insert(entity);
+            return new TableTransactionAction(TableTransactionActionType.Add, entity);
         }
 
-        internal async Task<IEnumerable<DynamicTableEntity>> ExecuteQuerySafeAsync(CloudTable metricsTable, TableQuery query)
+        internal async Task<IEnumerable<TableEntity>> ExecuteQuerySafeAsync(TableClient metricsTable, Expression<Func<TableEntity, bool>> query)
         {
             try
             {
-                return await ExecuteQueryWithContinuationAsync(metricsTable, query);
+                List<TableEntity> results = new List<TableEntity>();
+
+                await foreach (var result in metricsTable.QueryAsync(query))
```

Comment:
So the manual paging we were doing previously is no longer required?

Reply:
Based on conversations with @jviau the new SDK should paginate automatically. See https://github.com/Azure/azure-functions-host/pull/10167#discussion_r1604009472.

Reply:
Yes - the track 2 SDKs have custom `IAsyncEnumerable` implementations that will handle pagination for you.